typegate/tests/utils/test.ts

```diff
@@ -260,39 +202,68 @@ export class MetaTest {
     return engine;
   }
 
-  async engineFromTgDeployPython(path: string, cwd: string) {
+  async engine(path: string, opts: ParseOptions = {}) {
     const extension = extname(path);
     let sdkLang: SDKLangugage;
     switch (extension) {
       case ".py":
         sdkLang = SDKLangugage.Python;
         break;
+      case ".ts":
+      case ".js":
+      case ".mjs":
+        sdkLang = SDKLangugage.TypeScript;
+        break;
       default:
         throw new Error(`Unsupported file type ${extension}`);
     }
 
-    const serialized = await this.#serializeTypegraphFromShell(
+    const testDirName = dirname(path);
+    const cwd = join(testDir, testDirName);
+
+    const serialized = await this.#deployTypegraphFromShell(
       path,
       sdkLang,
       cwd,
+      opts,
     );
 
-    return await this.engineFromDeployed(serialized);
+    return await this.#engineFromDeployed(serialized, opts.secrets ?? {});
   }
 
-  async #serializeTypegraphFromShell(
+  async #deployTypegraphFromShell(
     path: string,
     lang: SDKLangugage,
     cwd: string,
+    opts: ParseOptions,
   ): Promise<string> {
-    // run self deployed typegraph
+    let output;
+    const secrets = opts.secrets ?? {};
+    const secretsStr = JSON.stringify(secrets);
+
+    if (lang === SDKLangugage.TypeScript) {
+      output = await this.shell([
+        lang.toString(),
+        "run",
+        "--allow-all",
+        "utils/tg_deploy_script.ts",
+        cwd,
+        this.port.toString(),
+        path,
+        secretsStr,
+      ]);
+    } else {
+      output = await this.shell([
+        lang.toString(),
+        "utils/tg_deploy_script.py",
+        cwd,
+        this.port.toString(),
+        path,
+        secretsStr,
+      ]);
+    }
 
-    const { stderr, stdout } = await this.shell([
-      lang.toString(),
-      path,
-      cwd,
-      this.port.toString(),
-    ]);
+    const { stderr, stdout } = output;
 
     if (stderr.length > 0) {
```

Comment:
Just run into this. Using stderr for error detection is not a good idea is unconventional and it ought to rely on exit codes alone. The typegraph implementation should avoid writing to stderr by default anyways and definitely not to signal errors.

Reply:
It's actually the deploy script that's writing to stderr.